.github/workflows/tests.yml

```
@@ -0,0 +1,53 @@
+name: Run all tests
+
+# Run this workflow every time a new commit is pushed to the repository
+on: push
```

This won't work if people open a PR from forks.

```suggestion
on:
  push:
    branches:
    - master
  pull_request:
```

If you want pushes to any branch to trigger a build, remove the filter. The setup above avoids duplicate build in PRs from branches in this repo.